src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -162,11 +162,11 @@ public Task StartAsync()
 
                     await _startTcs.Task;
                     await _receiveLoopTask;
-
                     _logger.LogDebug("Draining event queue");
-                    await _eventQueue.Drain();
 
-                    _httpClient.Dispose();
+                    await Task.WhenAny(
```

Comment:
So I actually started with exactly what you suggested. I copied over the NoThrowAwaiter from [here](https://github.com/aspnet/SignalR/blob/9a53598d8a9a77d676f9d1726f1a45edb60725be/src/Microsoft.AspNetCore.Sockets.Http/Internal/NoThrowTaskExtensions.cs) but you can't pass that into Task.WhenAny.